hub/tasks.py

```diff
@@ -194,15 +214,23 @@ async def import_pages(
             current_page=current_page + 1,
             request_id=request_id,
         )
-    else:
-        return await signal_request_complete.configure(
-            # Dedupe `refresh_pages` jobs for the same config
-            # https://procrastinate.readthedocs.io/en/stable/howto/queueing_locks.html
-            queueing_lock=f"request_complete_{request_id}"
-        ).defer_async(
-            request_id=request_id,
-            external_data_source_id=external_data_source_id,
-        )
+
+    # mark batch of jobs as completed
+    enqueue_result = await signal_request_complete.configure(
+        # Dedupe `refresh_pages` jobs for the same config
+        # https://procrastinate.readthedocs.io/en/stable/howto/queueing_locks.html
+        queueing_lock=f"request_complete_{request_id}"
+    ).defer_async(
+        request_id=request_id,
+        success=not error,
+        external_data_source_id=external_data_source_id,
+    )
+
+    # raise any error to mark this job as failed
+    if error:
+        raise error
+
+    return enqueue_result
```

Comment:
I wonder if all this `signal_request_complete` filter/checking stuff can be replaced by a new boolean field on the new `BatchRequest` model. Wdyt @joaquimds? Only worth it now if it improves robustness else maybe a backlog item.